js/client-chat.js

```diff
@@ -185,10 +185,14 @@
 		// highlight
 
 		getHighlight: function (message) {
-			var highlights = Tools.prefs('highlights') || [];
-			if (!app.highlightRegExp) {
+			var highlights = Tools.prefs('highlights') || {};
+			if (highlights.constructor === 'Array') this.convertHighlights();
+			var thisroom = (this.type === 'battle' ? 'battlehls' : this.id);
+			if (!app.highlightRegExp || !app.highlightRegExp[(Config.server.id + 'globalhls')] || !app.highlightRegExp[(Config.server.id + thisroom)]) {
 				try {
-					this.updateHighlightRegExp(highlights);
+					if (!app.highlightRegExp) app.highlightRegExp = [];
+					if (!app.highlightRegExp['globalhls']) this.updateHighlightRegExp(highlights[(Config.server.id + 'globalhls')], 'globalhls');
```

Comment:
^ are you referring to how i update a serverid + global key even though im calling a global key from highlightregexp? if so it will be fixed in the next commit.
would you prefer if i use just the roomid/global/battle as the highlightregexp key or should i use serverid + roomid/global/battle to keep it the same as the actual hls?